controllers/pkg/cluster/cluster.go

```diff
@@ -21,20 +21,30 @@ import (
 	"strings"
 
 	"github.com/nephio-project/nephio/controllers/pkg/cluster/capi"
+	"github.com/nephio-project/nephio/controllers/pkg/cluster/openshift"
 	"github.com/nephio-project/nephio/controllers/pkg/resource"
 	corev1 "k8s.io/api/core/v1"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+const kubeconfig string = "kubeconfig"
+
 type Cluster struct {
 	client.Client
 }
 
 func (r Cluster) GetClusterClient(secret *corev1.Secret) (ClusterClient, bool) {
-	switch string(secret.Type) {
-	case "cluster.x-k8s.io/secret":
-		if strings.Contains(secret.GetName(), "kubeconfig") {
+	if strings.Contains(secret.GetName(), kubeconfig) {
+		switch string(secret.Type) {
+		case "cluster.x-k8s.io/secret":
 			return &capi.Capi{Client: r.Client, Secret: secret}, true
+		case "Opaque":
+			// In case of OpenShift, the secret will have the following name
+			// $clusterName-admin-kubeconfig and the label hive.openshift.io/secret-type: kubeconfig
+			val, ok := secret.Labels["hive.openshift.io/secret-type"]
+			if ok && val == kubeconfig {
+				return &openshift.OpenShift{Client: r.Client, Secret: secret}, true
+			}
 		}
```

Comment:
Is there another approach here rather than hard coding the Openshift specific call? Perhaps a way of configuring this so that the label and the function to call is read from config/metadata would be a better approach.